### ISSUE TITLE
Relative Orientation in RoadPosition

### ISSUE THREAD

**Issue by hbanzhaf** [esmini/esmini#340]
I am specifying a trajectory with RoadPositions. Each entry looks like this:
```
<RoadPosition roadId="5" s="57.59949364977589" t="0.1777983370589864">
        <Orientation h="-0.024275732818746842" p="0" r="0" type="relative"/>
</RoadPosition>
```

esmini requires the heading angle (relative) to be computed like this:
```
    heading_offset = np.pi if lane_id > 0 else 0
    heading_relative = world_position.orientation.h - heading_road + heading_offset
```

Is this compliant with OSC1.2, where it says [here](https://www.asam.net/static_downloads/ASAM_OpenSCENARIO_V1.2.0_Model_Documentation/modelDocumentation/content/Orientation.html): 
"..., the Heading=0 means the Heading is aligned with the direction of the reference lane or road s-axis."

**Comment by eknabevcc** [comment]
I don't fully understand the issue. Can you further specify the problem? Perhaps you can also share a scenario that indicates the issue.

In [OSC 1.2 RoadPosition](https://www.asam.net/static_downloads/ASAM_OpenSCENARIO_V1.2.0_Model_Documentation/modelDocumentation/content/RoadPosition.html) it says:
"The relative reference context refers to the orientation with respect to the (s,t)-coordinate system of the target road that is considered a reference orientation."

So to me it makes sense that a specified relative heading is relative the road direction (s-axis). For example:

- relative h=0.0 is perfectly along the road direction (s-axis)
- relative h=0.1 is pointing slightly counter-clockwise from s-axis
- relative h=3.14 (pi) is opposite direction from s-axis

And all this disregarding the actual (world) direction of the road.

**Comment by eknabevcc** [comment]
Now I suddenly realized your point. We had an old issue (#228) which reported similar problems with RelativeRoadPosition.

In the end we decided to make relative heading depend on driving direction in RelativeLanePosition, while not in RelativeRoadPosition. For good and bad (more info in #228).

It makes sense for RoadPosition to behave in same way, i.e. relative heading should not be affected on driving direction (side of the road). Just as you expect and propose!

It has now been fixed by commit f9b0c1dc10f0913755d799d58805ee5ae6a7b4e7.

Please test and report if it works as expected or not.

**Comment by hbanzhaf** [comment]
Thanks @eknabevcc for making these changes. I can only test them if you provide me the binaries (have not set up the build pipeline yet).

**Comment by eknabevcc** [comment]
Here's an updated binary (currently [daaf01d](https://github.com/esmini/esmini/commit/daaf01d845bc3c6854ae5a92c6b09c15efa30c60)) for Windows. I hope that works. Let me know if you need for Linux instead.

**Comment by hbanzhaf** [comment]
I would need it for Linux. Thanks a lot!

**Comment by eknabevcc** [comment]
Commit is included in release [v2.26.9](https://github.com/esmini/esmini/releases/tag/v2.26.9)
So you can get the binary from there.

**Comment by hbanzhaf** [comment]
Looks good, thanks for integrating!
